### PR TITLE
Metadata override

### PR DESCRIPTION
**Pull request by cfkanesan**
## Purpose

Provide the functionality to update the attached GRIB message with relevant metadata

## Code changes

- Added `idpi.metadata.override` function

### CODE REVIEW

File: src/idpi/metadata.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        Keyword arguments forwarded to earthkit-data GribMetadata override method
+
+    Returns
+    -------
+    dict[str, Any]
+        Updated message byte string along with the geography and parameter namespaces
+
+    """
+    stream = io.BytesIO(message)
+    [grib_field] = ekd.from_source("stream", stream)
```
Comment (petrabaumann) [inline src/idpi/metadata.py:29]: I think that this works, as long as you do not have to use another product definition template to code the result.

**Review by petrabaumann — COMMENTED** [review]
If I interpret the code correctly, you create a new GRIB2 message by creating it from that of the parent, i.e. using the parent's GRIB2 record as a sample. This mechanism will not work, if you have to use a new product definition template to code the result of an operation, e.g., computing the ensemble mean from member data.

**Comment by cfkanesan** [comment]
> If I interpret the code correctly, you create a new GRIB2 message by creating it from that of the parent, i.e. using the parent's GRIB2 record as a sample. This mechanism will not work, if you have to use a new product definition template to code the result of an operation, e.g., computing the ensemble mean from member data.

Couldn't you update the relevant keys to reflect the ensemble mean?

**Comment by petrabaumann** [comment]
Depending on the product definition template number, different keys have to be set.

**Review by victoria-cherkas — APPROVED** [review]
Looks good to me. As @petrabaumann says it might be that you need to override metadata in a particular order eg product definition number first and then other keys that come with that template. But i think this can be implemented in another PR when we need to modify such metdata, eg statistics.

**Comment by victoria-cherkas** [comment]
If you leave that for another PR, perhaps you can add a comment to the override docstring explaining that the function is designed for modifications within the same template.

**Pull request merged**